SatisfactoryModLoader/util/Utility.cpp

```diff
@@ -5,10 +5,57 @@
 #include "DetoursFwd.h"
 #include "Utility.h"
 #include <game/Utility.h>
-#include <windows.h>
 #include <util/EnvironmentValidity.h>
+#include <Windows.h>
+#include <detours.h>
+
+#include "Objects/UFunction.h"
+#include "Objects/UClass.h"
+#include "Objects/FFrame.h"
+#include "Objects/FMemory.h"
 
 namespace SML {
+	namespace Objects {
+		bool DataCompare(PBYTE pData, PBYTE bSig, const char* szMask) {
+			for (; *szMask; ++szMask, ++pData, ++bSig) {
+				if (*szMask == 'x' && *pData != *bSig)
+					return false;
+			}
+			return (*szMask) == 0;
+		}
+
+		DWORD_PTR FindPattern(DWORD_PTR dwAddress, DWORD dwSize, const char* pbSig, const char* szMask, long offset) {
+			size_t length = strlen(szMask);
+			for (size_t i = NULL; i < dwSize - length; i++) {
+				if (DataCompare((PBYTE)dwAddress + i, (PBYTE)pbSig, szMask))
+					return dwAddress + i + offset;
+			}
+			return 0;
+		}
+
+		void initObjects() {
+			DWORD_PTR BaseAddress = (DWORD_PTR)GetModuleHandle(NULL);
+
+			MODULEINFO ModuleInfo;
+			GetModuleInformation(GetCurrentProcess(), (HMODULE)BaseAddress, &ModuleInfo, sizeof(ModuleInfo));
+
+			auto GNamesAddress = FindPattern(BaseAddress, ModuleInfo.SizeOfImage,
+				"\x48\x8B\x05\x00\x00\x00\x03\x48\x85\xC0\x0F\x85\xB0", "xxx???xxxxxxx", 0);
+			auto GNamesOffset = *reinterpret_cast<uint32_t*>(GNamesAddress + 3);
+			Objects::FName::names = *reinterpret_cast<Objects::TNameEntryArray**> (GNamesAddress + 7 + GNamesOffset);
+
+			auto GObjectsAddress = FindPattern(BaseAddress, ModuleInfo.SizeOfImage,
+				"\x48\x8D\x0D\x00\x00\x00\x04\xE8\x00\x00\x00\x00\xE8\x00\x00\x00\x00\xE8\x00\x00\x00\x01", "xxx???xx???xx???xx???x", 0);
+			auto GObjectsOffset = *reinterpret_cast<uint32_t*>(GObjectsAddress + 3);
+			Objects::UObject::objs = reinterpret_cast<Objects::FUObjectArray*>(GObjectsAddress + 7 + GObjectsOffset);
+
+			FName::fNameConstruct_f = (void (WINAPI*)(void*, const wchar_t*, FName::EFindName)) DetourFindFunction("FactoryGame-Win64-Shipping.exe", "FName::FName");
+			UClass::findFunction_f = (UFunction* (WINAPI*)(UObject*, FName)) DetourFindFunction("FactoryGame-Win64-Shipping.exe", "UObject::FindFunction");
+		
+			FMemory::init();
+		}
+	}
+
```

Comment:
Why are you duplicating already existing functions?

https://github.com/satisfactorymodding/SatisfactoryModLoader/blob/master/SatisfactorySDK/SDK.hpp#L1790-L1835

Reply:
Cause @SuperCoder7979 told me to try to make the new objects used for BPI independent from the SDK... I can try to change it though ^^

Reply:
Nope, that's fine